common/client_config_v2.go

```diff
@@ -24,6 +26,13 @@ type ClientConfigV2 struct {
 	PutTries                   int
 	MaxBlobSizeBytes           uint64
 	EigenDACertVerifierAddress string
+
+	// Fields required for validator payload retrieval
+	BLSOperatorStateRetrieverAddr string
+	EigenDAServiceManagerAddr     string
+
+	// RetrieversToEnable specifies which retrievers should be enabled
+	RetrieversToEnable []RetrieverType
```

Comment:
do we plan on having more than 2 types? Is there any reason why people would not always want both? Perhaps just to not have to setup all the contract addresses required to setup the validator retriever?

Reply:
> Is there any reason why people would not always want both

For testing purposes, it's really nice to have either one or the other. For production, it's currently not even an option for users to opt out of distributed retrieval (by design). 

> do we plan on having more than 2 types?

Ideally, I think that we should eventually define the various caching / fallback "backends" in this way. I don't see any reason why they couldn't conform to the disperser / retriever interfaces, and then configuration would be a lot more straightforward. It was with this plan in mind that I decided to pursue this more dynamic strategy

Reply:
Nice I think that would indeed be a really nice code refactor